### ISSUE TITLE
Output flag appears non-functional

### ISSUE THREAD

**Issue by KyleTryon** [bats-core/bats-core#678]
**Describe the bug**
Attempting to write the results of the tests to a file, I noticed there is a flag (`-o, --output`) which states `Directory to write report files`. Setting this flag results in no files being created after running tests.

It may be that I misunderstand the purpose of this flag, as it does not seem to offer any method of naming the files.

**To Reproduce**
1. `bats src/tests -o /tmp/bats;`
2. Run `ls /tmp/bats`
3. The directory is empty.

**Expected behavior**
A file named `results.xml` to be created in the output directory (in the case of the junit formatter)

**Environment (please complete the following information):**
 - Bats Version 1.8.2
 - OS: Linux/MacOS
 - Bash version: 3.2

**Comment by martin-schulze-vireso** [comment]
The `-o` flag on its own is useless because it only affects the `--report-formatter` output. If you don't specify one, no output will be generated. I think this is a shortcoming in the docs and maybe we could even add a check for that.

**Comment by martin-schulze-vireso** [comment]
Yes that sounds like a different issue, please provide a reproducer in the new issue.

**Comment by luishernandezv** [comment]
@pismy 

I was having issues with empty report files when using docker to run bats tests. Not sure if this was your issue but will just add here in case someone bumps into this issue. 

Suppose you have a folder structure like the following:

```bash
.
├─ test 
      ├─ test.bats
```
And you are running tests like this:

Replace `YOUR_DOCKER_IMAGE`
```bash
docker run -it -v `pwd`:`pwd` -w `pwd` YOUR_DOCKER_IMAGE sh -c 'bats -r test/bats --report-formatter junit --output `pwd`'
```

This would render an empty report file.

Now if you add a sleep to the end of the command 

```bash
docker run -it -v `pwd`:`pwd` -w `pwd` YOUR_DOCKER_IMAGE sh -c 'bats -r test/bats --report-formatter junit --output `pwd`; sleep 5'
```

This would output a proper report file.

Not sure of the internals of bats but it seems that the process to write the reports is separate from the main process which allows docker to exit without waiting for the report to be written.

**Comment by martin-schulze-vireso** [comment]
@luishernandezv that sounds like it is worth its own issue!
